src/clib/pio_msg.c

```diff
@@ -378,11 +399,20 @@ int inq_att_handler(iosystem_desc_t *ios)
         return pio_err(ios, NULL, PIO_ENOMEM, __FILE__, __LINE__);
     if ((mpierr = MPI_Bcast((void *)name, namelen + 1, MPI_CHAR, ios->compmaster,
                             ios->intercomm)))
+    {
+        free(name);
         return check_mpi2(ios, NULL, mpierr, __FILE__, __LINE__);
+    }
     if ((mpierr = MPI_Bcast(&xtype_present, 1, MPI_CHAR, 0, ios->intercomm)))
+    {
+        free(name);
         return check_mpi2(ios, NULL, mpierr, __FILE__, __LINE__);
+    }
     if ((mpierr = MPI_Bcast(&len_present, 1, MPI_CHAR, 0, ios->intercomm)))
+    {
+        free(name);
         return check_mpi2(ios, NULL, mpierr, __FILE__, __LINE__);
+    }
```

Comment:
Too much duplication of code. Instead, change to:

if (!mpierr)
    mpierr = MPI_Bcast(...)

Then at the end of all the calls
if (mpierr)
{
   free(name);
   check_mpi2(...);
}